### PR TITLE
Restart mdm service when new certificate is downloaded

### PR DESCRIPTION
**Pull request by petrkotas**
### What this PR does / why we need it:

When systemd downloader downloads fresh certificate
check whether it differs from the stored one.

Replace old one with fresh when there is a difference.

Signed-off-by: Petr Kotas <pkotas@redhat.com>

### Which issue this PR addresses:

Fixes https://msazure.visualstudio.com/AzureRedHatOpenShift/_workitems/edit/9882010/

### Test plan for issue:

Manual test for certificate change and service restart.

### Is there any documentation that needs to be updated for this PR?

No

### CODE REVIEW

**Comment by petrkotas** [comment]
I have deployed full RP in test environment, rotated mdm and mdsd certificate. I confirm mdm container rotating. The mdsd should reread the certificate automatically. However it does not report it in the logs:
```
[root@rp-8498a3d-000000 cloud-user]# journalctl -u mdsd.service
-- Logs begin at Tue 2021-08-24 15:22:55 UTC, end at Wed 2021-08-25 13:27:10 UTC. --
Aug 24 15:23:07 rp-8498a3d-000000 systemd[1]: Starting Azure MDS daemon (on systemd)...
Aug 24 15:23:07 rp-8498a3d-000000 mdsd[1504]: Starting Azure MDS Daemon: [  OK  ]
Aug 24 15:23:07 rp-8498a3d-000000 systemd[1]: Started Azure MDS daemon (on systemd).
```

File: pkg/deploy/devconfig.go [hunk 1]
```diff
@@ -105,6 +105,8 @@ func DevConfig(_env env.Core) (*Config, error) {
 			},
 		},
 		Configuration: &Configuration{
+			// AzureCloudName:               to.StringPtr("AzureCloud"),
+			// KeyvaultDNSSuffix:            to.StringPtr("vault.azure.net"),
```
[the remark this answers is not in the document]
Comment (petrkotas) [reply pkg/deploy/devconfig.go]: No we do not, thanks for pointing it out. I forgot about it.

**Comment by petrkotas** [comment]
@m1kola as we have discussed I added the documentation for the cert rotation

File: docs/certificate-rotation.md [hunk 2]
```diff
@@ -0,0 +1,59 @@
+# Certificate rotation
+
+First party certificate rotation for
+
+- RP,
+- MDSD,
+- MDM,
+
+is implemented in three different places.
```
Comment (m1kola) [inline docs/certificate-rotation.md:9]: Just a suggestion for better readability.

```suggestion
First party certificate rotation for the following components is implemented in three different places:

- RP
- MDSD
- MDM
```

**Review by m1kola — DISMISSED** [review]

**Comment by m1kola** [comment]
@petrkotas changes look good to me, but I think after updating `pkg/deploy/generator/resources_gateway.go` you did not run `make generate` to update bindata.

**Review by m1kola — APPROVED** [review]

**Review by gvanderpotte — APPROVED** [review]
LGTM

**Pull request merged**